src/types/linestring.rs

```diff
@@ -52,6 +53,16 @@ impl<T: WktNum> LineString<T> {
         let dim = coords[0].dimension();
         Self::new(coords, dim)
     }
+
+    /// Return the dimension of this geometry.
+    pub fn dimension(&self) -> Dimension {
+        self.dim
+    }
+
+    /// Consume self and return the inner parts.
+    pub fn into_inner(self) -> (Vec<Coord<T>>, Dimension) {
```

Comment:
This is a good one to have.

How would you feel about also adding `line_string.coords()` so that people can still borrow the field like they used to. (and similarly for the other variants)

Then the only thing missing in terms of functionality from the previous version with public fields would be `coords_mut()`, but since it seems less likely to be used, I'm not as concerned about adding it.

Reply:
Presumably this would be `coords(&self) -> &[Coord<T>]`.

How do you feel about methods with the same name across the struct and a trait that have slightly different signatures? E.g. if `LineStringTrait` is in scope, then `coords()` might use the implementation from that trait, which returns an iterator but _not_ necessarily a slice.

Reply:
Naming collisions are annoying, but I don't see a good way to avoid it in this case.

Reply:
So you're saying we should still add the `coords(&self) -> &[Coord<T>]` method? I'm fine with that.

Reply:
Correct!